src/sentry/static/sentry/app/views/settings/organizationDeveloperSettings/sentryApplicationRowButtons.tsx

```diff
@@ -0,0 +1,223 @@
+            showDelete
+            onPublish={onClickPublish}
+            onDelete={onClickRemove}
+            disablePublishReason={disablePublishReason}
+            disableDeleteReason={disableDeleteReason}
+          />
+        );
+      }}
+    </Access>
+  );
+};
+
+type UninstallButtonProps = {
+  install: SentryAppInstallation;
+  app: SentryApp;
+  onClickUninstall?: (install: SentryAppInstallation) => void;
+};
+const UninstallButton = ({install, app, onClickUninstall}: UninstallButtonProps) => {
+  const message = t(`Are you sure you want to remove the ${app.slug} installation?`);
+
+  return (
+    <Confirm
+      message={message}
+      priority="danger"
+      onConfirm={() => onClickUninstall && install && onClickUninstall(install)}
+    >
+      <StyledButton borderless icon="icon-trash" data-test-id="sentry-app-uninstall">
+        {t('Uninstall')}
+      </StyledButton>
+    </Confirm>
+  );
+};
+
+type InstallButtonProps = {
+  onClickInstall?: () => void;
+};
+const InstallButton = ({onClickInstall}: InstallButtonProps) => {
+  return (
+    <Button
+      onClick={onClickInstall}
+      size="small"
+      icon="icon-circle-add"
+      className="btn btn-default"
+    >
+      {t('Install')}
+    </Button>
+  );
+};
+
+type ActionButtonsProps = {
+  org: Organization;
+  app: SentryApp;
+
+  showDashboard: boolean;
+  showPublish: boolean;
+  showDelete: boolean;
+  onPublish?: () => void;
+  onDelete?: (app: SentryApp) => void;
+  // If you want to disable the publish or delete buttons, pass in a reason to display to the user in a tooltip
+  disablePublishReason?: string;
+  disableDeleteReason?: string;
+};
+
+const ActionButtons = ({
+  org,
+  app,
+  showDashboard,
+  showPublish,
+  showDelete,
+  onPublish,
+  onDelete,
+  disablePublishReason,
+  disableDeleteReason,
+}: ActionButtonsProps) => {
+  const appDashboardButton = showDashboard ? (
+    <StyledButton
+      size="small"
+      icon="icon-stats"
+      to={`/settings/${org.slug}/developer-settings/${app.slug}/dashboard/`}
+    >
+      {t('Dashboard')}
+    </StyledButton>
+  ) : null;
+
+  const publishRequestButton = showPublish ? (
+    <StyledButton
+      disabled={!!disablePublishReason}
+      title={disablePublishReason}
+      icon="icon-upgrade"
+      size="small"
+      onClick={onPublish}
+    >
+      {t('Publish')}
+    </StyledButton>
+  ) : null;
+
+  const deleteConfirmMessage = t(
+    `Deleting ${app.slug} will also delete any and all of its installations. \
+       This is a permanent action. Do you wish to continue?`
+  );
+  const deleteButton = showDelete ? (
+    disableDeleteReason ? (
+      <StyledButton
+        disabled
+        title={disableDeleteReason}
+        size="small"
+        icon="icon-trash"
+        label="Delete"
+      />
+    ) : (
+      <ConfirmDelete
+        message={deleteConfirmMessage}
+        confirmInput={app.slug}
+        priority="danger"
+        onConfirm={() => onDelete && onDelete(app)}
```

Comment:
@nolachen It would be better to not even render a delete button if we didn't have `onDelete`